airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -823,7 +817,7 @@ def _is_in_period(input_date: datetime) -> bool:
             if "Contents" in page:
                 new_keys = page["Contents"]
                 if _apply_wildcard:
-                    new_keys = (k for k in new_keys if fnmatch.fnmatch(k["Key"], _original_prefix))
```

Comment:
I son't think that is correct to use `Path` on non FS related objects, like S3 Key. In additional actual type object depend on OS.

```python
from pathlib import Path
import fnmatch

sample_s3_keys = {
    "/foo/bar/spam/egg",  # this should be invalid key because s3 should not be started with `/`
    "foo/bar/spam/egg",
    "foo////bar///spam/egg"
}

for key in sample_s3_keys:
    path_key = Path(key)
    print(f"{type(path_key).__name__}, {path_key}, {path_key.match('foo/bar*')}")
    print(f"{type(key).__name__}, {key}, {fnmatch.fnmatch(key, 'foo/bar*')}")
    print()
```

Linux/macOS output
```console
PosixPath, /foo/bar/spam/egg, False
str, /foo/bar/spam/egg, False

PosixPath, foo/bar/spam/egg, False
str, foo/bar/spam/egg, True

PosixPath, foo/bar/spam/egg, False
str, foo////bar///spam/egg, False
```

Windows
```console
WindowsPath, \foo\bar\spam\egg, False
str, /foo/bar/spam/egg, False

WindowsPath, foo\bar\spam\egg, False
str, foo/bar/spam/egg, True

WindowsPath, foo\bar\spam\egg, False
str, foo////bar///spam/egg, False
```